The cleanup script stopped working

See https://github.com/zhihan/event-ledger/actions/runs/22376339223/job/64767304958#step:5:45

 python -m cleanup
  shell: /usr/bin/bash -e {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.12.12/x64
    PKG_CONFIG_PATH: /opt/hostedtoolcache/Python/3.12.12/x64/lib/pkgconfig
    Python_ROOT_DIR: /opt/hostedtoolcache/Python/3.12.12/x64
    Python2_ROOT_DIR: /opt/hostedtoolcache/Python/3.12.12/x64
    Python3_ROOT_DIR: /opt/hostedtoolcache/Python/3.12.12/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.12.12/x64/lib
    GCS_BUCKET: 
    GOOGLE_APPLICATION_CREDENTIALS: 
    GOOGLE_CLOUD_PROJECT: living-memories-488001
    LIVING_MEMORY_FIRESTORE_DATABASE: living-memories-db
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/home/runner/work/event-ledger/event-ledger/src/cleanup.py", line 60, in <module>
    main()
  File "/home/runner/work/event-ledger/event-ledger/src/cleanup.py", line 54, in main
    deleted_ids = cleanup_firestore(today)
                  ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/event-ledger/event-ledger/src/cleanup.py", line 36, in cleanup_firestore
    deleted_pairs = firestore_storage.delete_expired(today)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/event-ledger/event-ledger/src/firestore_storage.py", line 96, in delete_expired
    db = _get_client()
         ^^^^^^^^^^^^^
  File "/home/runner/work/event-ledger/event-ledger/src/firestore_storage.py", line 28, in _get_client
    return firestore.Client(**kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.12/x64/lib/python3.12/site-packages/google/cloud/firestore_v1/client.py", line 96, in __init__
    super(Client, self).__init__(
  File "/opt/hostedtoolcache/Python/3.12.12/x64/lib/python3.12/site-packages/google/cloud/firestore_v1/base_client.py", line 146, in __init__
    super(BaseClient, self).__init__(
  File "/opt/hostedtoolcache/Python/3.12.12/x64/lib/python3.12/site-packages/google/cloud/client/__init__.py", line 340, in __init__
    Client.__init__(
  File "/opt/hostedtoolcache/Python/3.12.12/x64/lib/python3.12/site-packages/google/cloud/client/__init__.py", line 197, in __init__
    credentials, _ = google.auth.default(scopes=scopes)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.12/x64/lib/python3.12/site-packages/google/auth/_default.py", line 752, in default
    raise exceptions.DefaultCredentialsError(_CLOUD_SDK_MISSING_CREDENTIALS)
google.auth.exceptions.DefaultCredentialsError: Your default credentials were not found. To set up Application Default Credentials, see https://cloud.google.com/docs/authentication/external/set-up-adc for more information.
Error: Process completed with exit code 1.

--- a/.github/workflows/cleanup.yml
+++ b/.github/workflows/cleanup.yml
@@ -11,6 +11,11 @@ jobs:
     steps:
       - uses: actions/checkout@v4
 
+      - id: auth
+        uses: google-github-actions/auth@v2
+        with:
+          credentials_json: ${{ secrets.GCP_CREDENTIALS_JSON }}
+
       - uses: actions/setup-python@v5
         with:
           python-version: '3.12'
@@ -22,6 +27,5 @@ jobs:
         run: python -m cleanup
         env:
           GCS_BUCKET: ${{ secrets.GCS_BUCKET }}
-          GOOGLE_APPLICATION_CREDENTIALS: ${{ secrets.GCP_SA_KEY_PATH }}
           GOOGLE_CLOUD_PROJECT: living-memories-488001
           LIVING_MEMORY_FIRESTORE_DATABASE: living-memories-db

